jamf_protect_api/scripts/python/send_teams_webhook_analytics.py

```diff
@@ -0,0 +1,190 @@
+
+    createdDate = datetime.strptime(created, "%Y-%m-%dT%H:%M:%S.%fZ")
+    updatedDate = datetime.strptime(updated, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+    analytic_source = "Jamf" if jamf else "Custom Analytic"
+
+    payload = {
+        "@type": "MessageCard",
+        "summary": "Jamf Protect Analytic",
+        "sections": [
+            {
+                "activityTitle": f"{analyticType} Jamf Protect Analytic",
+                "activitySubtitle": f"{PROTECT_INSTANCE.upper()}",
+                "facts": [
+                    {"name": "Name", "value": f"{name}"},
+                    {"name": "Severity", "value": f"{sev}"},
+                    {"name": "Owner", "value": f"{analytic_source}"},
+                    {"name": "Description", "value": f"{desc}"},
+                    {
+                        "name": "Created",
+                        "value": f"{createdDate.strftime('%Y-%m-%d %H:%M')}",
+                    },
+                    {
+                        "name": "Modified",
+                        "value": f"{updatedDate.strftime('%Y-%m-%d %H:%M')}",
+                    },
+                ],
+                "markdown": "true",
+            }
+        ],
+        "potentialAction": [
+            {
+                "@type": "OpenUri",
+                "name": "View Analytic in Jamf Protect",
+                "targets": [
+                    {
+                        "os": "default",
+                        "uri": f"https://{PROTECT_INSTANCE}.protect.jamfcloud.com/analytics/{uuid}",
+                    }
+                ],
+            }
+        ],
+    }
+    teams_resp = requests.post(TEAMS_URL, json=payload)
+    print(teams_resp.status_code)
+
+
+def __main__():
+
+    # Get the access token
+    access_token = get_access_token(PROTECT_INSTANCE, CLIENT_ID, PASSWORD)
+
+    # Get all available Analytics
+    resp = make_api_call(PROTECT_INSTANCE, access_token, LIST_ANALYTICS_QUERY)
+
+    jamf_analytics_dict = {a["hash"]: a for a in resp["data"]["listAnalytics"]["items"]}
+
+    # Check if json file exists, if not, create and dump json
+    if (
+        os.path.exists("/var/tmp/jamf-protect-analytics-update-teams-webhook.json")
+        == False
+    ):
+        print("Creating JSON file")
+        with open(
+            "/var/tmp/jamf-protect-analytics-update-teams-webhook.json", "w"
+        ) as output:
+            json.dump(jamf_analytics_dict, output)
+
+    output = json.load(
+        open("/var/tmp/jamf-protect-analytics-update-teams-webhook.json", "r")
+    )
+
+    # Check diff betweetn json json file and new pull
+    new_jamf_analytics = set(jamf_analytics_dict).difference(output)
+
+    if not new_jamf_analytics:
+        print(f"Nothing to see here.")
+        return
+
+    print(f"New analytics available.")
+    for a_hash in new_jamf_analytics:
+        new_analytic = jamf_analytics_dict[a_hash]
+        if any(new_analytic["uuid"] in d.values() for d in output.values()):
```

Comment:
What you have will probably work 99.9999999% of the time but seems more straightforward to compare the uuids specifically rather than all the values.
```suggestion
        if any(new_analytic["uuid"] == old_analytic["uuid"] for old_analytic in output.values()):
```

Reply:
fixed